content/spin/plugin-authoring.md

```diff
@@ -0,0 +1,41 @@
+
+This document will cover what Spin plugins are, how to use a plugin, and how to contribute a plugin.
+
+## What are Spin plugins?
+
+A Spin plugin is an executable that is added to Spin's plugins directory (`$XDG_DATA_HOME/spin/plugins`) upon a `spin plugins install <plugin-name>`. The plugin is then ready to be used. If the plugin is an extension to the Spin CLI, it can now be executed directly as a subcommand: `spin <plugin-name>`. If the plugin is a trigger plugin, it will be executed during `spin up` when an app using that trigger is run. 
+
+While for now plugins are assumed to be executables, in the future, support for plugging in WebAssembly modules may be desirable.
+
+## How to find and use a Spin plugin
+
+Spin maintains a [Spin plugins repository](https://github.com/fermyon/spin-plugins) that acts as an inventory of available plugins. The `spin plugins` command can be used to list plugins (available and installed) and install a new plugin. For example, the `js2wasm` plugin is needed in order to write JavaScript Spin applications. It can be installed from the Spin plugins repository, using the following command: 
+
+```sh
+spin plugins install js2wasm
+```
+
+During `spin build`, `spin js2wasm` is called to build the JS Spin application. Learn more about building Spin components in JavaScript [here](https://developer.fermyon.com/spin/javascript-components.md).
+
+## Authoring a Spin plugin
+
+As mentioned previously, Spin plugins are executables that are executed either directly via the Spin CLI or as a trigger. 
+
+Creating and using a plugin can be summarized as follows:
+1. Build, package, and zip the plugin
+1. Create a Spin plugin manifest for the plugin
+1. Install the plugin (by pointing to the plugin manifest with the `--file` or `--url` flag)
+
+As step two suggests, a Spin plugin is defined by a Spin plugin manifest, which is a JSON file that conforms to the [Spin plugin manifest schema](https://github.com/fermyon/spin-plugins/blob/main/json-schema/spin-plugin-manifest-schema-0.1.json). A plugin manifest defines a plugin’s name, version, license, homepage (i.e. GitHub repo), compatible Spin version, and gives a short description of the plugin. It also points to the plugin source for various operating systems and platforms. When installing a plugin (via `spin plugin install`), by default, Spin will search the [`spin-plugins` repository](https://github.com/fermyon/spin-plugins/tree/main/manifests) for the plugin. The `--file` or `--url` flags can be used to point to the local or remote location of the Spin plugin manifest of a newly developed plugin.
```

Comment:
"Source" could be confusing here - it's the URL of the tar archive right?

Reply:
yes, a url but it can have a file scheme such as `"file:///tmp/binary.tgz"`